spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -131,7 +130,27 @@ public SubscriberFactory getSubscriberFactory() {
 	}
 
 	@Override
-	public void ack(Collection<String> ackIds, String subscriptionName) {
+	public void ack(Collection<AcknowledgeablePubsubMessage> acknowledgeablePubsubMessages) {
+		groupAcknowledgeableMessages(acknowledgeablePubsubMessages).forEach(this::ack);
```

Comment:
I think we need to add `Assert.notEmpty()`. Otherwise NPE in the `groupAcknowledgeableMessages()`.